### PR TITLE
openvino-dev uses opencv-python-headless as default

### PR DESCRIPTION
**Pull request by eaidova**
### Details:
 - *switch openvino-dev package on opencv-python-headless version by default due to problem to run tools in docker and server platforms without UI*

### Tickets:
 - *88523*

### CODE REVIEW

**Review by ilya-lavrenov — APPROVED** [review]

**Comment by ilya-lavrenov** [comment]
should we port to 2022.2 as well?

**Comment by eaidova** [comment]
> should we port to 2022.2 as well?

if we need these changes in 2022.2, then yes I can port

**Comment by ilya-lavrenov** [comment]
> > should we port to 2022.2 as well?
> 
> if we need these changes in 2022.2, then yes I can port

Why not? :) the change is useful

**Comment by eaidova** [comment]
> Why not? :) the change is useful

ok, here PR with port to releases/2022/2 https://github.com/openvinotoolkit/openvino/pull/12559

**Comment by ilya-lavrenov** [comment]
For some reason wheel steps are failed on CI, could you please have a look whether is it connected with dependency change?

**Comment by eaidova** [comment]
> For some reason wheel steps are failed on CI, could you please have a look whether is it connected with dependency change?

As I can see failed pip-conflicts on ubuntu20 due to different specified numpy version in nncf (numpy~=1.19.2, whlie in other ov components numpy>=1.16.6,<=1.23.1). It is not connected with my changes, it was introduced by @rkazants in https://github.com/openvinotoolkit/openvino/pull/12191 (probably pip-conflicts was disabled for his PR)
cc @akladiev

@AlexKoff88 are you ok to update numpy requirements for nncf?

**Comment by AlexKoff88** [comment]
I think we can relax NNCF numpy requirements and change them to numpy>=1.19.2 if it works for the rest of the components.

File: tools/benchmark_tool/setup.py [hunk 1]
```diff
@@ -21,6 +23,13 @@
         in pkg_resources.parse_requirements(requirements_txt)
     ]
 
+try:
+    importlib.import_module('cv2')
+except ImportError as opencv_import_error:
+    warnings.warn(
+            "Problem with cv2 import: \n{}\n opencv-python-headless will be added to requirements".format(opencv_import_error)
+    )
+    reqs.append('opencv-python-headless==4.5.*')
```
Comment (rkazants) [inline tools/benchmark_tool/setup.py:32]: Why don't we support the latest stable version 4.6?

File: tools/benchmark_tool/setup.py [hunk 2]
```diff
@@ -21,6 +23,13 @@
         in pkg_resources.parse_requirements(requirements_txt)
     ]
 
+try:
+    importlib.import_module('cv2')
+except ImportError as opencv_import_error:
+    warnings.warn(
+            "Problem with cv2 import: \n{}\n opencv-python-headless will be added to requirements".format(opencv_import_error)
```
Comment (rkazants) [inline tools/benchmark_tool/setup.py:30]: May be better to say: `will be installed`

File: tools/benchmark_tool/setup.py (hunk 1 above)
Comment (eaidova) [reply tools/benchmark_tool/setup.py:32]: I do not know, I'm not benchmark app developer :) 
I just replace one lib to another with keeping version as was used before

File: tools/benchmark_tool/setup.py (hunk 2 above)
Comment (eaidova) [reply tools/benchmark_tool/setup.py:30]: we do not install it in this place, just add to requirements file and keep user informed that we do it

Comment (eaidova) [reply tools/benchmark_tool/setup.py:30]: P.S. Please don't make me change this if there is not strong reason as I will have to align it with the message inside AC (merge one more PR in omz), update submodule and then again ,create PR for releases/2022/2. It is painful for me and require a lot of time for precommit checks :)

File: tools/benchmark_tool/setup.py (hunk 1 above)
Comment (rkazants) [reply tools/benchmark_tool/setup.py:32]: I think you can check it with the latest version and relax upper bound

Comment (eaidova) [reply tools/benchmark_tool/setup.py:32]: I do not want mixup dependencies management and resolving issue with running on headless platform, the first one is not my responsibility.
If you need it, then please create ticket

Comment (eaidova) [reply tools/benchmark_tool/setup.py:32]: But if you need my opinion, I would say that here should not be binding specific opencv version at all for benchmark app (only basic functionality used - read image, resize with linear interpolation, which not affected by new releases).
Anyway this does not help me to pass precommit... :(

Comment (rkazants) [reply tools/benchmark_tool/setup.py:32]: What is with pre-commit?

**Comment by rkazants** [comment]
> > For some reason wheel steps are failed on CI, could you please have a look whether is it connected with dependency change?
> 
> As I can see failed pip-conflicts on ubuntu20 due to different specified numpy version in nncf (numpy~=1.19.2, whlie in other ov components numpy>=1.16.6,<=1.23.1). It is not connected with my changes, it was introduced by @rkazants in #12191 (probably pip-conflicts was disabled for his PR) cc @akladiev
> 
> @AlexKoff88 are you ok to update numpy requirements for nncf?

What is wrong with `numpy>=1.16.6,<=1.23.1`? Could you please elaborate it more? I would like us to have the same version of numpy in all requirement files.

**Review by ilyachur — APPROVED** [review]

**Pull request merged**